app/src/main/java/com/github/onlynotesswent/model/users/UserRepositoryFirestore.kt

```diff
@@ -96,11 +97,32 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepositor
   }
 
   override fun deleteUserById(id: String, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(id)
-        .delete()
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception -> onFailure(exception) }
+    getUserById(
+        id,
+        { user ->
+          // for each of the current user follower remove them from their following list
+          user.friends.followers.forEach { follower ->
+            Log.e("RemovingMyselfFromFollower", follower)
+            db.collection(collectionPath)
+                .document(follower)
+                .update("friends.following", FieldValue.arrayRemove(id))
+          }
+          // for each of the current user following remove them from their follower list
+          user.friends.following.forEach { following ->
+            Log.e("RemovingMyselfFromFollowing", following)
```

Comment:
Like above, there is no need to log each time the action is performed. After testing, it can be removed